ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Byron.hs

```diff
@@ -40,49 +40,59 @@ instance RunNode ByronBlock where
 
   -- The epoch size is fixed and can be derived from @k@ by the ledger
   -- ('kEpochSlots').
-  nodeEpochSize           = \_proxy cfg _epochNo -> return
-                              . (coerce :: EpochSlots -> EpochSize)
-                              . kEpochSlots
-                              . Genesis.gdK
-                              . extractGenesisData
-                              $ cfg
+  nodeEpochSize             = \_proxy cfg _epochNo -> return
+                                . (coerce :: EpochSlots -> EpochSize)
+                                . kEpochSlots
+                                . Genesis.gdK
+                                . extractGenesisData
+                                $ cfg
+
+  nodeMaxBlockSize          = Aux.getMaxBlockSize . byronLedgerState
+  nodeBlockEncodingOverhead =
+    let blockHeaderOverhead = 650 {- the maximum block header size -}
+        blockBodyOverhead   = 1 {- ABody: encodeListLen 4 -}
+                            + 1 {- TxPayload: list -}
+                            + 1 {- Delegation.Payload: list -}
```

Comment:
BTW, a list has in fact an overhead of 2 bytes (I might have said 1 to you at some point):
```haskell
import           Codec.Serialise
import           Codec.CBOR.Write
import qualified Data.ByteString.Lazy as Lazy

nbBytes :: Serialise a => a -> Int
nbBytes = fromIntegral . Lazy.length . toLazyByteString . encode

> nbBytes True
1
> nbBytes [True]
3
> nbBytes [True, True]
4
> nbBytes (replicate 1000 True)
1002